test/language/statements/async-function/syntax-declaration-no-line-terminator.js

```diff
@@ -0,0 +1,14 @@
+// Copyright 2016 Microsoft, Inc. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Brian Terlson <brian.terlson@microsoft.com>
+es7id: pending
+description: Async function declarations cannot have a line break after `async`
+info: Reference error is thrown due to looking up async in strict mode
+flags: [onlyStrict]
```

Comment:
I think the ReferenceError occurs outside of strict mode

>     ExpressionStatement : Expression ;
>
>     1. Let exprRef be the result of evaluating Expression.
>     2. Return ? GetValue(exprRef). 

and

>     6.2.3.1 GetValue (V)
>
>     1. ReturnIfAbrupt(V).
>     2. If Type(V) is not Reference, return V.
>     3. Let base be GetBase(V).
>     4. If IsUnresolvableReference(V) is true, throw a ReferenceError exception.

which is distinct from

>     PutValue (V, W)
>
>     1. ReturnIfAbrupt(V).
>     2. ReturnIfAbrupt(W).
>     3. If Type(V) is not Reference, throw a ReferenceError exception.
>     4. Let base be GetBase(V).
>     5. If IsUnresolvableReference(V) is true, then
>        a. If IsStrictReference(V) is true, then
>             i. Throw a ReferenceError exception.
>        b. Let globalObj be GetGlobalObject().
>        c. Return ? Set(globalObj, GetReferencedName(V), W, false).